ccf_transaction_processor/transaction_processor/pdo_tp.cpp

```diff
@@ -137,15 +166,145 @@ namespace ccfapp
                 HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Enclave already registered");
             }
 
-            // Verify enclave data
-            string proof_data = in.proof_data;
-            if (proof_data.empty()) {
-                // Enclave proof data is empty - simulation mode
-            }else{
+            //Determine if enclave attestation check is enabled or not
+            auto attestation_policy_view = ctx.tx.rw(attestation_policy_table);
+
+            // First ensure that any policy has been set. Else, fail the rpc.
+            auto attestation_policy_global = attestation_policy_view->get_globally_committed(PDO_ENCLAVE_ATTESTATION_POLICY);
+            if (!attestation_policy_global.has_value()){
                 return ccf::make_error(
-                    HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Only simulation mode is currently supported");
+                    HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "No attestation policy has been set. Enclave cannot be registered");
             }
 
+            auto attestation_policy=attestation_policy_global.value();
+
+            if(attestation_policy.check_attestation){
+
+                //ensure that proof data is not empty
+                if(in.proof_data.empty()) {
+                  return ccf::make_error(
+                      HTTP_STATUS_BAD_REQUEST, ccf::errors::InvalidInput, "Proof data cannot be empty while attestation check is enabled");
+                }
+
+                /* the following items are verified
+
+                1. ias report signature
+                2. epid pseudonym
+                3. enclave quote status
+                4. MREnclave
+                5. nonce
+                6. basename
+                7. user report data
+
+                */
+
```

Comment:
Great! Nice to see this work in the CCF TP

Reply:
thank you for all your help!